### ISSUE TITLE
console input, 2x GPU and enable_model_cpu_offload()

### ISSUE THREAD

**Issue by Yakonrus** [huggingface/diffusers#5091]
### Describe the bug

I am using a script with an infinite loop and console input.
pipe.enable_model_cpu_offload(gpu_id=1)
The second time the proc() function is executed, the model is uploaded to GPU 0, although only GPU 1 is set to be used.

![Screenshot 2023-09-18 211026](https://github.com/huggingface/diffusers/assets/4978662/6c9fc918-7641-454f-8d17-42b2673b4690)
![Screenshot 2023-09-18 211033](https://github.com/huggingface/diffusers/assets/4978662/138f1bd5-87a9-47eb-be53-cb3dc0bf4a18)

### Reproduction

	#!/usr/bin/env python3

	CUDA_VISIBLE_DEVICES=1

	import string
	import os
	from sys import exit
	import torch
	from PIL import Image

	from diffusers import StableDiffusionXLPipeline
	base_text_pipe = StableDiffusionXLPipeline.from_pretrained(
		"stabilityai/stable-diffusion-xl-base-1.0",
		torch_dtype=torch.float16,
		variant="fp16",
		use_safetensors=True,
		add_watermarker=False,
	)
	base_text_pipe.enable_model_cpu_offload(gpu_id=1)

	def proc():

		image = base_text_pipe(
			prompt="Hulk",
			negative_prompt="cartoon",
			num_inference_steps=10,
		).images[0]
		
		image.convert("RGB").save("0.jpg", quality=100, subsampling=0) 

	while True:
		try:
			print("Waiting for a task")
			ans = input("")
			proc();
			
		except Exception as e:
			print(e)
			
	exit(0)  

### System Info

Windows 11 22H2
python 3.10.11

Package                      Version
---------------------------- ------------------------
absl-py                      1.4.0
accelerate                   0.23.0
addict                       2.4.0
albumentations               1.3.1
antlr4-python3-runtime       4.9.3
astunparse                   1.6.3
attrs                        23.1.0
av                           10.0.0
basicsr                      1.4.2
beautifulsoup4               4.12.2
black                        23.9.1
blinker                      1.6.2
cachetools                   5.3.1
certifi                      2022.12.7
cffi                         1.15.1
chardet                      5.2.0
charset-normalizer           2.1.1
click                        8.1.7
cloudpickle                  2.2.1
colorama                     0.4.6
coloredlogs                  15.0.1
contourpy                    1.1.1
controlnet-aux               0.0.7
cycler                       0.11.0
Cython                       3.0.2
deepface                     0.0.79
detectron2                   0.6
detectron2-densepose         0.6
diffusers                    0.22.0.dev0
dlib                         19.24.2
easydict                     1.10
einops                       0.6.1
facexlib                     0.3.0
filelock                     3.9.0
filterpy                     1.4.5
fire                         0.5.0
Flask                        2.3.3
flatbuffers                  23.5.26
fonttools                    4.42.1
fsspec                       2023.4.0
future                       0.18.3
fvcore                       0.1.5.post20221221
gast                         0.4.0
gdown                        4.7.1
google-auth                  2.23.0
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
grpcio                       1.58.0
gunicorn                     21.2.0
h5py                         3.9.0
huggingface-hub              0.17.1
humanfriendly                10.0
hydra-core                   1.3.2
idna                         3.4
imageio                      2.31.3
importlib-metadata           6.8.0
insightface                  0.7.3
iopath                       0.1.9
itsdangerous                 2.1.2
Jinja2                       3.1.2
joblib                       1.3.2
keras                        2.13.1
kiwisolver                   1.4.5
lazy_loader                  0.3
libclang                     16.0.6
llvmlite                     0.40.1
lmdb                         1.4.1
Markdown                     3.4.4
MarkupSafe                   2.1.2
matplotlib                   3.8.0
mediapipe                    0.10.5
mpmath                       1.2.1
mtcnn                        0.1.1
mypy-extensions              1.0.0
networkx                     3.0rc1
numba                        0.57.1
numpy                        1.24.1
oauthlib                     3.2.2
omegaconf                    2.3.0
onnx                         1.14.1
onnxruntime                  1.15.1
opencv-contrib-python        4.8.0.76
opencv-python                4.8.0.76
opencv-python-headless       4.8.0.76
opt-einsum                   3.3.0
packaging                    23.1
pandas                       2.1.0
pathspec                     0.11.2
Pillow                       9.3.0
pip                          23.2.1
platformdirs                 3.10.0
portalocker                  2.8.2
prettytable                  3.9.0
protobuf                     3.20.3
psutil                       5.9.5
pyasn1                       0.5.0
pyasn1-modules               0.3.0
pycocotools                  2.0.7
pycparser                    2.21
pyparsing                    3.1.1
pyreadline3                  3.4.1
PySocks                      1.7.1
pyTelegramBotAPI             4.13.0
python-dateutil              2.8.2
pytz                         2023.3.post1
PyWavelets                   1.4.1
pywin32                      306
PyYAML                       6.0.1
qudida                       0.0.4
regex                        2023.8.8
requests                     2.28.1
requests-oauthlib            1.3.1
retina-face                  0.0.13
rsa                          4.9
safetensors                  0.3.3
scikit-image                 0.21.0
scikit-learn                 1.3.0
scipy                        1.11.2
seaborn                      0.12.2
setuptools                   65.5.0
six                          1.16.0
sounddevice                  0.4.6
soupsieve                    2.5
sympy                        1.11.1
tabulate                     0.9.0
tb-nightly                   2.15.0a20230915
telebot                      0.0.5
tensorboard                  2.13.0
tensorboard-data-server      0.7.1
tensorflow                   2.13.0
tensorflow-estimator         2.13.0
tensorflow-intel             2.13.0
tensorflow-io-gcs-filesystem 0.31.0
termcolor                    2.3.0
threadpoolctl                3.2.0
tifffile                     2023.8.30
timm                         0.9.7
tokenizers                   0.13.3
tomli                        2.0.1
torch                        2.2.0.dev20230916+cu121
torchaudio                   2.2.0.dev20230917+cu121
torchvision                  0.17.0.dev20230917+cu121
tqdm                         4.66.1
transformers                 4.33.2
typing_extensions            4.4.0
tzdata                       2023.3
urllib3                      1.26.13
wcwidth                      0.2.6
Werkzeug                     2.3.7
wheel                        0.41.2
wrapt                        1.15.0
yacs                         0.1.8
yapf                         0.40.1
zipp                         3.16.2

### Who can help?

@patrickvonplaten

**Comment by Yakonrus** [comment]
WSL2, ubuntu 22.04
Same situation

![Screenshot 2023-09-20 012453](https://github.com/huggingface/diffusers/assets/4978662/441034e0-2b0d-40c8-a14d-20fef31d90cf)
![Screenshot 2023-09-20 012500](https://github.com/huggingface/diffusers/assets/4978662/810da638-b8c9-4b1b-93f9-b0e9a985e5c6)

**Comment by Yakonrus** [comment]
@patrickvonplaten
What to do about it?
How to use only GPU 1?

**Comment by patrickvonplaten** [comment]
Hey @Yakonrus, 

Great catch! You're totally right. Can you check if this PR fixes it for you: https://github.com/huggingface/diffusers/pull/5132/files

**Comment by Yakonrus** [comment]
pip install -U git+https://github.com/huggingface/diffusers.git@533771fb7c34c4b864507210692046b9a2fece38
Works correctly

**Comment by frankjoshua** [comment]
For anyone else with the same issue. I also had to also remove a change to device ordering in addition to installing the patch.

Setting CUDA_DEVICE_ORDER to PCI_BUS_ID caused only gpu 0 to be usable with enable_model_cpu_offload()

Remove this:
```
import os
# Oder GPUs by PCI bus ID to match device numbering in nvidia-smi (before models are loaded)
os.environ["CUDA_DEVICE_ORDER"]="PCI_BUS_ID" 
```
I really only needed it when testing or debugging multiple gpus. So only a small loss to not have device order match nvidia-smi.
